### PR TITLE
Add mix format

### PR DESCRIPTION
**Pull request by josevalim**
All code in this PR has been formatted by the formatter.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,9 +22,12 @@ script:
...
+  - bin/elixir bin/mix format --dry-run --check-equivalent
   - dialyzer -pa lib/elixir/ebin --build_plt --output_plt elixir.plt --apps lib/elixir/ebin/elixir.beam lib/elixir/ebin/Elixir.Kernel.beam
 
 notifications:
   recipients:
-    - jose.valim@plataformatec.com.br
+    - jose.valim@gmail.com.br
     - eric.meadows.jonsson@gmail.com
+    - lexmag@me.com
+    - an.leopardi@gmail.com
```
Comment (whatyouhide) [inline .travis.yml]: 🖕 @fishcakez

File: lib/elixir/lib/code/formatter.ex [hunk 2]
```diff
@@ -79,7 +80,25 @@ defmodule Code.Formatter do
...
+    assert_received: 2,
+    refute: 1,
+    refute: 2,
+    refute_in_delta: 3,
+    refute_in_delta: 4,
+    refute_receive: 1,
+    refute_receive: 2,
+    refute_receive: 3,
+    refute_received: 1,
+    refute_received: 2
```
Comment (whatyouhide) [inline lib/elixir/lib/code/formatter.ex]: Oh boy I never pushed these D: I have them still on my local checkout of the formatter 😄

File: lib/mix/lib/mix/tasks/format.ex [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      reject contributions with unformatted code. However, keep in mind,
+      that the formatting output may differ between Elixir versions as
+      improvements and fixes are applied to the formatter.
+
+    * `--check-equivalent` - check if the file after formatting has the
+      same AST. If the ASTs are not equivalent, it is a bug in the code
+      formatter. This option is recommended if you are automatically
+      formatting files.
+
+    * `--dry-run` - do not save files after formatting
```
Comment (whatyouhide) [inline lib/mix/lib/mix/tasks/format.ex]: Missing trailing period here.

File: lib/mix/lib/mix/tasks/format.ex [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+    * `--print` - write formatted files to stdout instead of saving to disk
```
Comment (whatyouhide) [inline lib/mix/lib/mix/tasks/format.ex]: Missing trailing period here.

File: lib/mix/lib/mix/tasks/format.ex [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+    * `--dot-formatter` - the file with formatter configuration.
+      Defaults to `.formatter.exs` if one is available, see next section
```
Comment (whatyouhide) [inline lib/mix/lib/mix/tasks/format.ex]: Missing trailing period here.

File: .travis.yml (hunk 1 above)
Comment (josevalim) [reply .travis.yml]: He does not have a public e-mail listed on GitHub, so I will wait until he confirms if he wants that. :)

File: lib/mix/lib/mix/tasks/format.ex [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+  If any of the "check" flags are given and a check fails, the formatted
```
Comment (whatyouhide) [inline lib/mix/lib/mix/tasks/format.ex]: I would rewrite to "If any of the `--check-*` flags ...".

File: .travis.yml [hunk 7]
```diff
@@ -22,9 +22,12 @@ script:
   - make compile
   - rm -rf .git
   - make test
+  - bin/elixir bin/mix format --dry-run --check-equivalent
   - dialyzer -pa lib/elixir/ebin --build_plt --output_plt elixir.plt --apps lib/elixir/ebin/elixir.beam lib/elixir/ebin/Elixir.Kernel.beam
 
 notifications:
   recipients:
-    - jose.valim@plataformatec.com.br
+    - jose.valim@gmail.com.br
```
Comment (ericmj) [inline .travis.yml]: gmail.com

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: .travis.yml (hunk 1 above)
Comment (fishcakez) [reply .travis.yml]: I don't need to receive emails that he who must not be named has forced pushed again :rage:.

**Pull request merged**